src/main.rs

```diff
@@ -0,0 +1,124 @@
+use std::{fs::File, io::Write};
+
+use async_graphql::{
+    extensions::Logger, http::GraphiQLSource, EmptySubscription, SDLExportOptions, Schema,
+};
+use async_graphql_axum::{GraphQLRequest, GraphQLResponse};
+
+use authentication::AuthorizedUserHeader;
+use axum::{
+    extract::State,
+    http::{HeaderMap, StatusCode},
+    response::{self, IntoResponse},
+    routing::get,
+    Router, Server,
+};
+use clap::{arg, command, Parser};
+use s3::{creds::Credentials, Bucket, BucketConfiguration, Region};
+use simple_logger::SimpleLogger;
+
+use log::info;
+
+use crate::{mutation::Mutation, query::Query};
+
+mod mutation;
+mod query;
+
+mod media;
+mod media_connection;
+
+mod authentication;
+
+/// Builds the GraphiQL frontend.
+async fn graphiql() -> impl IntoResponse {
+    response::Html(GraphiQLSource::build().endpoint("/").finish())
+}
+
+/// Command line argument to toggle schema generation instead of service execution.
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// Generates GraphQL schema in `./schemas/media.graphql`.
+    #[arg(long)]
+    generate_schema: bool,
+}
+
+/// Connects to MinIO bucket if existent, otherwise creates bucket `media-data`.
+/// Warning: Change credentials before production use.
+async fn initialize_minio_media_data_bucket() -> Bucket {
+    let bucket_name = "media-data";
+    let region = Region::Custom {
+        region: "eu-central-1".to_string(),
+        endpoint: "http://media-minio:9000".to_string(),
```

Comment:
can this be hardcoded?
I realized that you also kept the `proxy_set_header Host media-minio:9000;` hardcoded in the fontend nginx
so if only these two need to match, that's fine
however if it has to match with the real one, this should be customizable for the k8s setup

Reply:
That should be a parsed environment variable, will fix that